warehouse/query-core/src/main/java/datawave/query/function/IndexOnlyContextCreator.java

```diff
@@ -199,6 +200,22 @@ protected DatawaveJexlContext newDatawaveJexlContext(final Tuple3<Key,Document,M
         return newContext;
     }
     
+    /**
+     * Ensures that downstream expectations are met with regard to the state of the given context
+     * 
+     * @param context
+     *            jexl context to verify, assumed to be non-null
+     */
+    private void verifyContext(DatawaveJexlContext context) {
+        // Make sure that index-only fields being queried are represented in the context, in
+        // order to avoid potential downstream eval misfire (see #1204)
+        for (String var : this.variables) {
+            if (this.indexOnlyFields.contains(var) && !context.has(var)) {
```

Comment:
@apmoriarty , @jwomeara it turns out that this method introduced the new issue (#1227) observed on our test system. That seems to be due to an existing shortcoming of **DelayedNonEventIndexContext** ...

That is, if the context passed in here is of type **DelayedNonEventIndexContext**, then `!context.has(var)` is _always_ true. That's because the **has** method isn't overridden to operate on the internal `delegate` context, unlike its **get** and **set** methods which do. It just interrogates the internal map owned by the instance itself, and that map is apparently always empty.

So, an index-only term that was previously fetched/added to the delegate context, and which would have constituted a match, is now being replaced in the delegate with an empty set...and the hit is lost

Is there a reason we shouldn't override the **has** method in **DelayedNonEventIndexContext** to operate on the delegate?

Reply:
Latest commit adds overrides for several methods to **DelayedNonEventIndexContext**